packages/react-dialog/Spec.md

```diff
@@ -2,62 +2,303 @@
+5. The dialog can be dismissed also when confirmation button is clicked (footer).
+
+#### Non-modal
+
+![Mouse and touch behaviour of a non-modal dialog](./assets/non-modal-mouse-touch.png)
+
+1. Clicking on the trigger (element / button component) a Dialog is displayed without a dimmed background.
+2. A user can continue to interact with elements on the page behind the dialog.
+3. Clicking the dismiss button (X icon in header), or cancel/dismiss buttons (footer) will close the dialog.
+
+#### Alert dialog
 
-- _Migration from v8_
-- _Migration from v0_
+![Mouse and touch behaviour of alert dialog](./assets/alert-mouse-touch.png)
 
-## Behaviors
+1. Clicking on the trigger (element / button component) a Dialog is displayed with a dimmed background.
+2. Widnows under the dialog are are inert, their scrolling is blocked and clicking on the dimmed background will not close the dialog.
+3. Clicking the dismiss button (X icon in header), or cancel/dismiss buttons (footer) will dismiss the dialog.
 
-_Explain how the component will behave in use, including:_
+### Keyboard
 
-- _Component States_
-- _Interaction_
-  - _Keyboard_
-  - _Cursor_
-  - _Touch_
-  - _Screen readers_
+#### Modal
+
+![Keyboard behaviour of a modal dialog](./assets/modal-keyboard.png)
+
+1. **(1)** TabKey to set focus on Trigger, use EnterKey to open.
+2. **(2-5)** Focus is moved to the first focusable control inside the dialog.
+3. **(5-6)** After the dialog is dismissed, keyboard focus should be moved back to where it was before it moved into the dialog. Otherwise the focus can be dropped to the beginning of the page. Or if the item is no longer available it can be moved to the next logical location in that region i.e. next / previous item.
+4. **TabKey** Moves focus to next focusable element inside the dialog. When focus is on the last focusable element in the dialog, moves focus to the first focusable element in the dialog.
+5. **Shift+Tab** Moves focus to previous focusable element inside the dialog. When focus is on the first focusable element in the dialog, moves focus to the last focusable element in the dialog.
+6. **EscKey** Closes the dialog.
+
+#### Non-modal
+
+![Keyboard behaviour of non-modal dialog](./assets/non-modal-keyboard.png)
+
+1. **(1)** **TabKey** to set focus on Trigger, use **EnterKey** to open.
+2. - **(2a)** Focus is moved to the default focusable control inside the dialog.
+   - **(2b)** **EnterKey** on dismiss action to close dialog,
+3. After the dialog is dismissed, keyboard focus should be moved back to where it was before it moved into the dialog. Otherwise the focus can be dropped to the beginning of the page.
+4. **TabKey** Moves focus to next focusable element inside the dialog, once you get to the end of the focusable items within the dialog focus moves to next actionable item outside of the dialog container.
+5. **Shift+Tab** Moves focus to previous focusable element inside the dialog and back to the trigger control.
```

Comment:
Is `EscKey` behavior be the same as `modal`?

Reply:
It will be the same if the focus is on the dialog, but if the focus is not on the dialog the esc will have a different behaviour depending on the content. I'll add this detail in the spec so it is addressed. Thnx